ginga/rv/plugins/Downloads.py

```diff
@@ -0,0 +1,203 @@
+        self.dlbox = Widgets.VBox()
+        fr.set_widget(self.dlbox)
+        vbox2.add_widget(fr, stretch=0)
+        vbox2.add_widget(Widgets.Label(''), stretch=1)
+        sc.set_widget(vbox2)
+        vbox.add_widget(sc, stretch=1)
+        self.w_scroll = sc
+
+        btns = Widgets.HBox()
+        btns.set_spacing(4)
+        btns.set_border_width(4)
+
+        btn = Widgets.Button("Close")
+        btn.add_callback('activated', lambda w: self.close())
+        btns.add_widget(btn)
+        btn = Widgets.Button("Remove All")
+        btn.add_callback('activated', lambda w: self.gui_remove_all())
+        btns.add_widget(btn, stretch=0)
+        btns.add_widget(Widgets.Label(''), stretch=1)
+        vbox.add_widget(btns, stretch=0)
+
+        container.add_widget(vbox, stretch=1)
+
+        self.gui_up = True
+
+    def gui_add_track(self, track):
+        vbox = Widgets.VBox()
+        fname = Widgets.TextEntry()
+        fname.set_text(track.info.url)
+        vbox.add_widget(fname, stretch=0)
+        hbox = Widgets.HBox()
+        hbox.set_spacing(2)
+        time_lbl = Widgets.Label(track.elapsed)
+        prog_bar = Widgets.ProgressBar()
+        prog_bar.set_value(track.progress)
+        hbox.add_widget(time_lbl)
+        hbox.add_widget(prog_bar)
+        rmv = Widgets.Button('Remove')
+
+        def _remove_download(w):
+            self.gui_rm_track(track)
+
+        rmv.add_callback('activated', _remove_download)
+        hbox.add_widget(rmv)
+        vbox.add_widget(hbox, stretch=0)
+        self.dlbox.add_widget(vbox)
+        w_track = Bunch.Bunch(container=vbox, time_lbl=time_lbl,
+                              prog_bar=prog_bar, track=track)
+        self.w_track[track.key] = w_track
+        self.w_scroll.scroll_to_end(vertical=True)
+
+    def gui_rm_track(self, track):
+        if track in self.downloads:
+            self.downloads.remove(track)
+
+        if self.gui_up:
+            w_track = self.w_track.get(track.key, None)
+            if w_track is not None:
+                del self.w_track[track.key]
+            self.dlbox.remove(w_track.container)
+
+    def gui_remove_all(self):
+        for track in list(self.downloads):
+            self.gui_rm_track(track)
+
+    def add_download(self, info, future):
+        add_time = time.time()
+        key = str(add_time)
+
+        # create tracker for this download
+        track = Bunch.Bunch(key=key, info=info, time_start=add_time,
+                            elapsed='00:00:00', progress=0.0,
+                            future=future)
+        self.downloads.append(track)
+
+        if self.gui_up:
+            self.gui_add_track(track)
+
+        self.fv.nongui_do(self.download, info.url, info.filepath, track)
+
+    def download(self, url, localpath, track):
+        def _dl_indicator(count, blksize, totalsize):
+            # calculate elapsed time label
+            cur_time = time.time()
+            elapsed = cur_time - track.time_start
```

Comment:
Is this not what you are trying to do?

```python
>>> from datetime import datetime
>>> d1 = datetime.now()
>>> # Wait a second or two
>>> d2 = datetime.now()
>>> dt = d2 - d1
>>> str(dt)
'0:00:04.000012'
```

Ref: https://stackoverflow.com/questions/538666/python-format-timedelta-to-string

Reply:
Why yes it is!  (didn't think to check the `str` of it).

Reply:
Although it looks like I might have to tweak the output a bit to get hours padded and to remove the fractional seconds.  I think I might leave this for a later fix.

Reply:
Weird that it is so hard to customize the formatting but maybe a suggestion from https://stackoverflow.com/questions/18470627/how-do-i-remove-the-microseconds-from-a-timedelta-object would work:

```python
str(dt).split(".")[0]
```

Reply:
> str(dt).split(".")[0]

Still need to pad hours, if we want a fixed size display.

Since I'm already subtracting two Python `time.time()` objects I am getting the result in seconds.  Not sure that the machinations that are necessary to force the `timedelta` object into the right format is prettier or more understandable than just dividing seconds by the necessary constants.  Plus, if the maintainers of `datetime` decide to change the `str` representation of `timedelta` objects then that way breaks, whereas the current scheme doesn't.

Reply:
That's true. I didn't realize formatting `timedelta` can get painful if its output format has to be customized. I thought for sure it is implemented already. Yeah, fine, let's stick with the current implementation then. Thanks for humoring me.